### PR TITLE
Update acuity-metadata and support shape inference for more operations

### PR DESCRIPTION
**Pull request by hawk081**
This PR contains two main changes:
1. Update acuity metadata
2. Support shape inference for more operations

### CODE REVIEW

File: source/acuity-metadata.json [hunk 1]
```diff
@@ -1,70 +1,235 @@
 [
   {
     "name": "a_times_b_plus_c",
+    "category": "Layer",
+    "attributes": [],
     "inputs": [
-      { "name": "A" },
-      { "name": "B" },
-      { "name": "C" }
+      { "name": "input0" },
```
Comment (lutzroeder) [inline source/acuity-metadata.json:7]: Use readable names for input names.

File: source/acuity-metadata.json [hunk 2]
```diff
@@ -1,70 +1,235 @@
...
+      { "name": "input1" },
+      { "name": "input2" }
     ],
+    "constants": [],
```
Comment (lutzroeder) [inline source/acuity-metadata.json:11]: Remove empty `"constants": []` and `"attributes: []`.

File: source/acuity-metadata.json [hunk 3]
```diff
@@ -1,70 +1,235 @@
...
     "outputs": [
       { "name": "output" }
     ]
   },
   {
     "name": "abs",
+    "category": "Layer",
```
Comment (lutzroeder) [inline source/acuity-metadata.json:18]: Follow the category patterns in other files. `Layer` is used Convolution, LSTM or Fully Connected layers.

**Review by lutzroeder — CHANGES_REQUESTED** [review]

**Comment by hawk081** [comment]
Updated.

**Comment by hawk081** [comment]
merged by 17b35c5014b51838b081cd4a47c96bfe47747925

**Pull request closed without merging**